fingertip/plugins/script.py

```diff
@@ -31,7 +24,41 @@ def run(m, scriptpath, cache=0, no_unseal=False):
         m.ssh.upload(scriptpath, dst=scriptname)
         m.log.debug(f'uploaded {scriptpath} into {scriptname}')
         m(f'chmod +x {scriptname}')
+        if mode == 'test':
+            m.hooks.disable_proxy()
+            check = False
+        elif mode == 'run':
+            check = True
+        else:
+            assert False
         m.log.plain()
-        m(f'./{scriptname}')
+        exit_code = m(f'./{scriptname}', check=check).retcode
         m.log.nicer()
+        m.log.debug(f'Exit code: {exit_code}')
     return m
+
+# execute setup-like scenario, meaning mostly
+#  * do not touch proxy settings
+#  * traceback on non-zero exit code
+#  * continue pipeline
+# script.run without --cache=X defaults to:
+#  * not caching if it is the last plugin in the pipeline
+#  * caching if there are further plugins in the pipeline
+# Examples:
+# fingertip ... + script.run script  -  no cache, never persist, always rerun
+# fingertip ... + script.run script + ssh  -  cache just for this invocation
+# fingertip ... + script.run script --cache=1h  -  try to cache for 1h at most
+# fingertip ... + script.run script --cache=1h + ssh  -  try reusing the cache
+# fingertip ... + transient script.run script --cache=1h + ssh  -  revert
+@fingertip.transient(when=_should_run_be_transient)
+def run(m, scriptpath, mode=None, cache=0, no_unseal=False):
+    return _exec(m, scriptpath, mode='run', cache=cache, no_unseal=no_unseal)
+
+# run test like scenario, meaning mostly:
+#  * disable proxy
+#  * do not traceback on non-zero exit code
+#  * finish the pipeline, report exit code
+#  * therefore, only makes sense as the last plugin in the pipeline
```

Comment:
why not `sys.exit` on nonzero retcode in all scenarios then? =)

Reply:
sorry, outdated comment, will update